### PR TITLE
add note about retrieving error stack to SSL_get_verify_result(3)

### PR DESCRIPTION
**Pull request by vladak**
This change augments the SSL_get_verify_result(3) man page with a note about retrieving error stack. This would have helped me when diagnosing a subtle certificate verification error.

### CODE REVIEW

**Review by paulidale — APPROVED** [review]

File: doc/man3/SSL_get_verify_result.pod [hunk 1]
```diff
@@ -22,6 +22,13 @@ of a certificate can fail because of many reasons at the same time. Only
 the last verification error that occurred during the processing is available
 from SSL_get_verify_result().
 
+Sometimes there can be a sequence of errors leading to the verification
+failure as reported by SSL_get_verify_result().
+To get the errors, it is necessary to setup a verify callback via
+L<SSL_CTX_set_verify(3)> or L<SSL_set_verify(3)> and retrieve the errors
+from the error stack there, because once L<SSL_connect(3)> returns,
+these errors mey no longer be available.
```
Comment (t8m) [inline doc/man3/SSL_get_verify_result.pod:30]: Typo: may

Comment (vladak) [reply doc/man3/SSL_get_verify_result.pod:30]: Fixed, thanks.

**Review by paulidale — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
Merges, thanks for the clarification.

**Comment by yhabteab** [comment]
Hi everyone, sorry for commenting on an already merged PR, but I'm trying to understand the [`SSL_get_verify_result(3)`](https://docs.openssl.org/master/man3/SSL_get_verify_result/#notes) man page notes added with this PR.

> Sometimes there can be a sequence of errors leading to the verification failure as reported by SSL_get_verify_result(). To get the errors, it is necessary to setup a verify callback via [SSL_CTX_set_verify(3)](https://docs.openssl.org/master/man3/SSL_CTX_set_verify/) or [SSL_set_verify(3)](https://docs.openssl.org/master/man3/SSL_CTX_set_verify/) and retrieve the errors from the error stack there, because once [SSL_connect(3)](https://docs.openssl.org/master/man3/SSL_connect/) returns, these errors may no longer be available.

The above notes says, *the errors may no longer be available once [`SSL_connect(3)`](https://docs.openssl.org/master/man3/SSL_connect/) returns*. Does that mean I won't be able to retrieve the cert verify error via `SSL_get_verify_result(3)` outside of the verify callback?

Also, noting that the errors might be gone after `SSL_connect(3)` returns, is that specific to client side handshake routines or does that apply to the server side as well?

I just want to know if I can safely use `X509_verify_cert_error_string(SSL_get_verify_result(ssl))` outside of the verify callback without losing the exact certification error.

Thanks!

**Pull request closed without merging**